pkg/clientaccess/token.go

```diff
@@ -250,18 +253,34 @@ func GetHTTPClient(cacerts []byte, certFile, keyFile string) *http.Client {
 	if err == nil {
 		tlsConfig.Certificates = []tls.Certificate{cert}
 	}
-
-	return &http.Client{
+	client := &http.Client{
 		Timeout: defaultClientTimeout,
 		Transport: &http.Transport{
 			DisableKeepAlives: true,
 			TLSClientConfig:   tlsConfig,
 		},
 	}
+
+	for _, o := range option {
+		o(client)
+	}
+	return client
+}
+
+func WithTimeout(d time.Duration) ClientOption {
+	return func(c *http.Client) {
+		c.Timeout = d
+	}
+}
+
+func WithHeaderTimeout(d time.Duration) ClientOption {
+	return func(c *http.Client) {
+		c.Transport.(*http.Transport).ResponseHeaderTimeout = d
+	}
```

Comment:
just a nit, but you could probably have WithTimeout set both timeouts, and I don't think anyone would object.

Reply:
That'd make the timestamps printed in the `apiserver has not reloaded encryption configuration` message more readable as well.